lib/api.js

```diff
@@ -110,6 +124,10 @@ function Api (plugins, defaultConfig) {
       }
 
       if (plug.name && typeof plug.name === 'string') {
+        if (plug.name === 'global') {
+          console.error('plugin named "global" is reserved, skipping')
+          return create
+        }
         const found = create.plugins.some((p) => p.name === plug.name)
```

Comment:
:fire: I think you need to compare camelized names

Test - create two plugins, one with name "my plugin", one with name "myPlugin". what happens to config passed to each? Does th "pluckOpts" pluck correctly

Perhaps names should be string with only letters and numbers (no spaces, dashes, underscores)

Reply:
This is old code, untouched by this PR, thus outside of the scope of this PR. This PR is not aiming at making everything better, it's only aiming at doing the config.global separation.